packages/angular_devkit/build_angular/src/utils/index-file/inline-fonts.ts

```diff
@@ -0,0 +1,170 @@
+
+      if (tagName !== 'link') {
+        return;
+      }
+
+      // <link tag with rel="stylesheet" and a href.
+      const href = attrs.find(({ name, value }) => name === 'rel' && value === 'stylesheet')
+        && attrs.find(({ name }) => name === 'href')?.value;
+
+      if (href) {
+        hrefList.push(href);
+      }
+    });
+
+    await new Promise(resolve => collectorStream.on('finish', resolve));
+
+    // Download stylesheets
+    const hrefsContent = await this.processHrefs(hrefList);
+    if (hrefsContent.size === 0) {
+      return content;
+    }
+
+    // Replace link with style tag.
+    const { rewriter, transformedContent } = await htmlRewritingStream(content);
+    rewriter.on('startTag', tag => {
+      const { tagName, attrs } = tag;
+
+      if (tagName !== 'link') {
+        rewriter.emitStartTag(tag);
+
+        return;
+      }
+
+      const hrefAttr = attrs.some(({ name, value }) => name === 'rel' && value === 'stylesheet')
+        && attrs.find(({ name, value }) => name === 'href' && hrefsContent.has(value));
+      if (hrefAttr) {
+        const href = hrefAttr.value;
+        const cssContent = hrefsContent.get(href);
+        rewriter.emitRaw(`<style type="text/css" title="${href}">${cssContent}</style>`);
+      } else {
+        rewriter.emitStartTag(tag);
+      }
+    });
+
+    return transformedContent;
+  }
+
+  private async getResponse(url: string, userAgent: UserAgent): Promise<string> {
+    const key = `${packageVersion}|${url}|${userAgent}`;
+
+    if (cacheFontsPath) {
+      const entry = await cacache.get.info(cacheFontsPath, key);
+      if (entry) {
+        return promisify(readFile)(entry.path, 'utf8');
+      }
+    }
+
+    const data = await new Promise<string>((resolve, reject) => {
+      let rawResponse = '';
+      https.get(
+        url,
+        {
+          headers: {
+            'user-agent': userAgent,
+          },
+        },
+        res => {
+          res
+            .on('data', chunk => rawResponse += chunk)
+            .on('end', () => resolve(rawResponse.toString()));
+        },
+      )
+        .on('error', e => reject(e));
+    });
+
+    if (cacheFontsPath) {
+      await cacache.put(cacheFontsPath, key, data);
+    }
+
+    return data;
+  }
+
+  private async processHrefs(hrefList: string[]): Promise<Map<string, string>> {
+    const hrefsContent = new Map<string, string>();
+
+    for (const href of hrefList) {
+      // Normalize protocols to https://
+      let normalizedHref = href;
+      if (!href.startsWith('https://')) {
+        if (href.startsWith('//')) {
+          normalizedHref = 'https:' + href;
+        } else if (href.startsWith('http://')) {
+          normalizedHref = href.replace('http:', 'https:');
+        } else {
+          // Unsupported CSS href.
+          continue;
+        }
+      }
+
+      if (!SUPPORTED_PROVIDERS.some(url => normalizedHref.startsWith(url))) {
+        // Provider not supported.
+        continue;
```

Comment:
We should probably print out a warning here

Reply:
Having unsupported providers is a valid use-case. One such example is for css files which are consumed via a CDN. Example: `https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css`